src/java.base/share/classes/java/lang/VersionProps.java.template

```diff
@@ -119,6 +124,12 @@ class VersionProps {
         props.put("java.vendor", VENDOR);
         props.put("java.vendor.url", VENDOR_URL);
         props.put("java.vendor.url.bug", VENDOR_URL_BUG);
+
+        // java.vm.vendor was set by VM, replace the value if customized
+        // Default branding.conf value is "N/A", which did not affect java.vm.vendor
+        if (! VENDOR_VM.equals("N/A")) {
+            props.put("java.vm.vendor", VENDOR_VM);
```

Comment:
Magnus might have suggestions on this, maybe it would be better to default in the conf file to the empty string so that we don't have to deal with "N/A" in several places.

Reply:
We can surely change the default. We just must make sure that all code that uses the vendor define is prepared to handle an empty string.

Reply:
I think the issue is not the default value, but the special treatment with default value. Is there a reason we didn't set COMPANY_NAME to "N/A" as in branding.conf?
https://github.com/openjdk/jdk/blame/master/make/autoconf/spec.gmk.template#L276-L285

Instead of default to "N/A", personally I think "OpenJDK" as an organization can be used?

Reply:
> Instead of default to "N/A", personally I think "OpenJDK" as an organization can be used?

OpenJDK is a source only distribution. As such vendor being set to OpenJDK for the binary doesn't make sense. It's the downstream distro shipping the binary that should set it. "N/A" is OK. I don't know if the empty string is a good one.

Reply:
Point taken.
I would imagine downstream vendor would modify branding.conf to proper value instead of using the default.

"N/A" is fine to me, question is whether COMPANY_NAME should be derived from branding.conf regardless.